Kudu.Services/ServiceHookHandlers/GithubHandler.cs

```diff
@@ -17,6 +36,25 @@ public override DeployAction TryParseDeploymentInfo(HttpRequestBase request, JOb
             return DeployAction.UnknownPayload;
         }
 
+        protected override string determineSecurityProtocol(JObject repository)
+        {
+            string repositoryUrl = repository.Value<string>("url");
+            bool isPrivate = repository.Value<bool>("private");
+            if (isPrivate)
+            {
+                repositoryUrl = repository.Value<string>("ssh_url");
```

Comment:
are you talking about writing 1 `GetSSHUrl` with implementation in the base class, or just a template method and have the children classes override them? either way its not going to be straight forward

[gitlab](https://github.com/projectkudu/kudu/blob/master/Kudu.Services/ServiceHookHandlers/GitlabHqHandler.cs#L82-L83),[codebase](https://github.com/projectkudu/kudu/blob/fd0062a0888bf54ad6f14f2d761e5947a57de983/Kudu.Services/ServiceHookHandlers/CodebaseHqHandler.cs#L38-L46) all have different logics to determine "private" repo ... actually they override `GetDeploymentInfo` 

so I can't have them override `GetSSHUrl` without refactor their `GetDeploymentInfo` function , I keep the old code because we also use `githubcompathandler` as one of the parser, I am not sure if there are cases that will match it...Moreover inside `githubhandler` I can check for "github.com" with confidence...I am thinking it's safer to use this white list kind of strategy (less assumption)